### PR TITLE
adding support for mouse scrolling

### PR DESCRIPTION
**Pull request by samdmarshall**
This adds basic support for scrolling with the mouse wheel, addressing #664.

The changes to ui.py on line 160 feels super hacky but it seems that the format of the events from urwid change based on if they are keyboard bindings or from a mouse. I think that should be able to handle all of the use cases (i wasn't able to get it to crash, and the tests passed after I ran `python setup.py test`).

### CODE REVIEW

File: alot/ui.py [hunk 1]
```diff
@@ -157,6 +157,8 @@ def fire(_, cmdline):
                     return [cmdline[5:]]
 
             key = keys[0]
+            if len(key) > 1 and key[0].startswith('mouse'):
```
Comment (dcbaker) [inline alot/ui.py]: you can simplify this:
`if key and key[0].starswith('mouse'):`

File: alot/ui.py [hunk 2]
```diff
@@ -157,6 +157,8 @@ def fire(_, cmdline):
...
+                key = key[0]+' %i' % key[1]
```
Comment (dcbaker) [inline alot/ui.py]: The formatting on this isn't quite right, we use spaces between operators.
so: `key = key[0] + ' %i' % key[1]`. Although I think that you can simplify this to `' '.join(key[:2])`

**Comment by dcbaker** [comment]
The changes look good, thanks for making them. I'd like to see the fixes squashed into the original patch, and then I'll be happy to merge it.

**Comment by samdmarshall** [comment]
How exactly would you like me to do that (squashing commits is not something do myself)?

**Comment by pazz** [comment]
Quoting Samantha Marshall (2017-01-09 20:25:00)
>    How exactly would you like me to do that (squashing commits is not
>    something do myself)?

You can use `git rebase -i` to do that, and force push to your PR branch
(it will update automatically here then).

I haven't tested this so just one question to make sure:
does this patch prevent you from selecting text from the terminal
screen? I vaguely remember that this was the reasoning behing removing
mouse support initially..

**Comment by dcbaker** [comment]
> I haven't tested this so just one question to make sure: does this patch prevent you from selecting text from the terminal screen? I vaguely remember that this was the reasoning behing removing mouse support initially..

I didn't test that. I will before we merge.

**Comment by dcbaker** [comment]
> I didn't test that. I will before we merge.

Okay, I've tested and you can in fact select text with this branch.

**Comment by samdmarshall** [comment]
Ok, I think this is ready to go.

**Comment by dcbaker** [comment]
Looks good, thanks!

**Comment by pazz** [comment]
@dcbaker which terminal are you using? I can now confirm that this in fact disables on-screen selection for me (urxvt). So unless we can separate mouse events 4/5 from the rest I personally would use the
setting proposed in #957 to disable this.

**Comment by lucc** [comment]
I can confirm that mouse scrolling works but selecting text **not** (needs shift pressed). In xterm, urxvt & konsole (all with and without tmux).

**Comment by pazz** [comment]
Quoting Lucas Hoffmann (2017-01-11 10:33:17)
>    I can confirm that mouse scrolling works but selecting text not (needs
>    shift pressed). In xterm, urxvt & konsole (all with and without tmux).

Uh! great find! I didn't know that it works with the shit key.
I could actually live with that..
Nevertheless, this should be configurable.

**Comment by lucc** [comment]
The shift key thing is a standard in all (?) terminal applications. Either you can select text and they do not support mouse events or you can not select text and they do support mouse events.

I am living with this with at least these programs: vim/neovim, tmux, tig, ncmpcpp

**Pull request merged**